src/world/besieged_system.cpp

```diff
@@ -36,31 +37,136 @@ bool BesiegedSystem::handleMessage(std::vector<uint8>&& payload,
                                    uint16               from_port)
 {
     // TODO: Handle incoming map messages
-    if (settings::get<bool>("logging.DEBUG_BESIEGED"))
-    {
-        ShowDebug(fmt::format("Message: unknown conquest type received from {}:{}",
+    DebugBesieged(fmt::format("Message: unknown conquest type received from {}:{}",
                               from_addr.s_addr,
                               from_port));
-    }
 
     return false;
 }
 
 void BesiegedSystem::updateVanaHourlyBesieged()
 {
-    auto strongholdInfos = this->getStrongholdInfos();
+    if (!settings::get<bool>("main.BESIEGED_ENABLED"))
+    {
+        return;
+    }
 
-    // Note to reviewer: logging.DEBUG_BESIEGED is not read properly from world.
-    // It is from map. Other settings in world seem to use main.conf
-    if (settings::get<bool>("logging.DEBUG_BESIEGED"))
+    updateBeastmenForces();
+    sendStrongholdInfosMsg();
+}
+
+void BesiegedSystem::updateBeastmenForces()
+{
+    // Iterate through all strongholds, excluding alzhabi, and update forces
+    // TODO: Iterating through enums is not a good practice and can be error prone
+    // For now, this is fine since these will not change often, but consider a better method
+    for (int i = BESIEGED_STRONGHOLD::MAMOOK; i <= BESIEGED_STRONGHOLD::ARRAPAGO; i++)
```

Comment:
Aware that this is bad, will refactor in a follow up before LSB merge. Open to suggestions on how to do it (it really is just a 3 value enum iteration).

Reply:
You could build an initializer list and iterate it:
```cpp
for (auto stronghold : { BESIEGED_STRONGHOLD::MAMOOK, BESIEGED_STRONGHOLD::WHATEVER, BESIEGED_STRONGHOLD::ARRAPAGO })
{
    ...
}
```